### ISSUE TITLE
Fix `declaration-property-value-no-unknown` false positives for `gradient(in color space)` of `background`

### ISSUE THREAD

**Issue by otomad** [stylelint/stylelint#8610]
### What minimal example or steps are needed to reproduce the bug?

```css
a {
	background: linear-gradient(in hsl longer hue, red, red);
	background: radial-gradient(in hsl longer hue, red, red);
	background: conic-gradient(in hsl longer hue, red, red);
	background: repeating-linear-gradient(in hsl longer hue, red, red 50px);
	background: repeating-radial-gradient(in hsl longer hue, red, red 50px);
	background: repeating-conic-gradient(in hsl longer hue, red, red 50px);
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
    "rules": {
        "declaration-property-value-no-unknown": true
    }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgIyTAGsBzAJwgFdMp4UAbAS0xiTIFpykpGZMcAFMxQALAM70GETCRhlRlGABoUZGFBVqoASgDcWXAWLkqNOmW6Mk9ThZ59Bw8ZPrTZ8kYs3rvO-dnxCUgpqWhRITEYwW0sHIUxRCSkZOQVlVR8MvwNA4xCzDIAHVhxmEnYmFjYY+354xJc3VM90rV8UAFYABkKADz0co2DTMLVipFKZdjsrGy5axwTnZPc033buvoGAoZNQ8xhxyfKIqJreOqck1xSPLyyNnv7-AF8QJRAAM0Z6GAA5JAAWzgiBgvSBhV+ADowGIxO9wNJviQECAMAkUJj0CAyJRfmJsXR0ZiSSTsbAwPQ2BNGNJ2IUKMUyDgAJ7sABu1kU7EwEHY1CIvIA7tgECgcLiYFhSS8sG8PhFkQAxCBkQETVEAKzE0gRsEK8MQxJQ2LErN+lRwhJNICpOBgZuxSmlNrNLItzBw7EVjHKZqQNDYUGt2LtDqtIDlIBeQA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "16.20.0"
}
```

### What did you expect to happen?

No problems to be reported.

### What actually happened?

It doesn't support interpolation color space feature (e.g. `in hsl`, `in oklch`, `in srgb`) in gradient functions.

However, it is interesting that `linear-gradient` is working fine and no error reported.

The following problems were reported:

```
example.css
3:14-58	error	Unexpected unknown value "radial-gradient(in hsl longer hue, red, red)" for property "background"	(declaration-property-value-no-unknown)
4:14-57	error	Unexpected unknown value "conic-gradient(in hsl longer hue, red, red)" for property "background"	(declaration-property-value-no-unknown)
5:14-73	error	Unexpected unknown value "repeating-linear-gradient(in hsl longer hue, red, red 50px)" for property "background"	(declaration-property-value-no-unknown)
6:14-73	error	Unexpected unknown value "repeating-radial-gradient(in hsl longer hue, red, red 50px)" for property "background"	(declaration-property-value-no-unknown)
7:14-72	error	Unexpected unknown value "repeating-conic-gradient(in hsl longer hue, red, red 50px)" for property "background"	(declaration-property-value-no-unknown)
```

### Do you have a proposal to fix the bug?

Update syntax in CSSTree.

**Comment by jeddy3** [comment]
Thanks for the report. As with https://github.com/stylelint/stylelint/issues/8609#issuecomment-2958582449, I'll mark as upstream.

**Comment by otomad** [comment]
For temporary use, you can extend the syntax yourself:

```json
{
	"languageOptions": {
		"syntax": {
			"types": { 
				"radial-gradient()": "| <color-interpolation-method>",
				"conic-gradient()": "| <color-interpolation-method>",
				"repeating-linear-gradient()": "| <color-interpolation-method>",
				"repeating-radial-gradient()": "| <color-interpolation-method>",
				"repeating-conic-gradient()": "| <color-interpolation-method>"
			}
		}
	}
}
```
